ideascube/mediacenter/forms.py

```diff
@@ -24,9 +27,38 @@ def save(self, commit=True):
         kind = guess_kind_from_content_type(content_type)
         if kind:
             document.kind = kind
-        if not document.id:
-            # m2m need the document to have an id.
+        if commit:
+            if not document.id:
+                # m2m need the document to have an id.
+                document.save()
+            self.save_m2m()
+            document.save()  # Search index needs m2m to be saved.
+        return document
+
+
+class NotFileDocumentForm(forms.ModelForm):
```

Comment:
Except that document does have a file.

It's just that the file is not uploaded through normal means, it already is in the right place and we just pass the file **path** to the form.

`PreuploadedDocumentForm`?